PPAK_CVS/Sources/Storage/Storage.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  Storage.swift
+//  PPAK_CVS
+//
+//  Created by 김응철 on 2022/12/25.
+//
+
+import Foundation
+
+final class Storage {
+  
+  static let shared = Storage()
+  
+  private let key = "Storage"
+  private let userDefaults = UserDefaults.standard
+  private init() {}
+  
+  private(set) var products: [ProductModel] = shared.load() {
+    didSet {
+      let encoder = JSONEncoder()
+      if let encoded = try? encoder.encode(oldValue) {
+        userDefaults.set(encoded, forKey: key)
+      }
+    }
+  }
+  
+  func load() -> [ProductModel] {
+    guard let data = userDefaults.object(forKey: key) as? Data else { return [] }
+    let decoder = JSONDecoder()
+    if let products = try? decoder.decode([ProductModel].self, from: data) {
+      return products
+    } else {
+      return []
+    }
```

Comment:
여기에서 사용되는 `JSONDecoder()`도 한 번만 사용되는 인스턴스네요. 바로 호출 및 접근하는 건 어떨까요?

```suggestion
    if let products = try? JSONDecoder().decode([ProductModel].self, from: data) {
      return products
    } else {
      return []
    }
```